src/pretix/base/models/items.py

```diff
@@ -897,3 +897,31 @@ def _position_lookup(self) -> Q:
 
     class QuotaExceededException(Exception):
         pass
+
+    @staticmethod
+    def clean_variations(items, variations):
+        for variation in variations:
+            if variation.item not in items:
+                raise ValidationError(_('All variations must belong to an item contained in the items list'))
```

Comment:
Could you end the error messages with a full stop, if they are a full sentence? :)